### PR TITLE
Update IDOL Data Types to Match Nova

### PR DESCRIPTION
**Pull request by riyajaggi**
### Summary 

This pull request updates the `member` and `role` data types for IDOL to match Nova's member jsons. The existing documents in the firebase are also updated to reflect the same changes.

- [x] change the case of member fields to camelCase
- [x] add missing fields (`roleDescription` and `netid`)
- [x] add missing roles (`'business'`)
- [x] update documents in firebase 

### Notes 

- Consider sharing .d.ts files with frontend for data type consistency

### Breaking Changes 

- Updated `Member` type: Check and update any frontend functionality for the same.

### CODE REVIEW

File: backend/src/DataTypes.d.ts [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+export type Role =
+  | 'lead'
+  | 'admin'
+  | 'tpm'
+  | 'pm'
+  | 'developer'
+  | 'designer'
+  | 'business';
+
+export type roleDescription =
```
Comment (SamChou19815) [inline backend/src/DataTypes.d.ts:12]: Can we standardize to use `PascalCase` here?

File: backend/data/run.ts [hunk 2]
```diff
@@ -27,21 +27,23 @@ db.collection('members')
...
+        about: jsonData.about,
         subteam: jsonData.subteam,
-        other_subteams: jsonData.otherSubteams,
-        website: jsonData.website
+        otherSubteams: jsonData.otherSubteams,
+        role: jsonData.roleId,
+        roleDescription: jsonData.roleDescription
       };
 
       removeEmptyOrNull(data);
```
Comment (SamChou19815) [inline backend/data/run.ts:49]: while you are here, I will point out that `removeEmptyOrNull` only removed undefined value, but not null values

**Review by SamChou19815 — CHANGES_REQUESTED** [review]
Changes LGTM with some small nits, but I will put a hold here since this will break our frontend. Now that we are in a monorepo, we can fix the frontend together as well. @riyajaggi do you want to make related frontend changes (I think it's mostly change the type def and fix casing) or I can do it?

File: backend/src/DataTypes.d.ts (hunk 1 above)
Comment (riyajaggi) [reply backend/src/DataTypes.d.ts]: done!

**Comment by riyajaggi** [comment]
> Changes LGTM with some small nits, but I will put a hold here since this will break our frontend. Now that we are in a monorepo, we can fix the frontend together as well. @riyajaggi do you want to make related frontend changes (I think it's mostly change the type def and fix casing) or I can do it?

Thanks for reviewing this PR! Yes, I added the changes to the `MembersAPI.ts` for frontend. If there's anything I missed for frontend, feel free to add.

File: backend/data/run.ts (hunk 2 above)
Comment (riyajaggi) [reply backend/data/run.ts:49]: Thank you for pointing this out! I believe this script was made by @henry-li-06 for initially adding/modifying jsons to the firebase. For this PR, I actually edited the script a little when I updated the documents in the firebase to take care of the undefined values and update fields. Should I push the updated script to git?

Comment (SamChou19815) [reply backend/data/run.ts:49]: Feel free to add it.

**Comment by SamChou19815** [comment]
@riyajaggi you need to fix the additional type errors in the frontend

**Comment by SamChou19815** [comment]
I fixed typing for now. But the frontend doesn't have support for role field yet. We should fix this in another PR.

**Review by SamChou19815 — APPROVED** [review]

**Pull request merged**